Refactor Events, SDK Bootstrapping, and 0x

pgebheim opened the pull request.

nuevoalex committed:
fix Gnosis tests

pgebheim committed:
Add helper to fetch addresses for one network

pgebheim committed:
Refactor ZeroX instance out of augursdk / Augur instances

pgebheim committed:
Use standard file naming for BaseConnector

pgebheim committed:
Refactor out passing configs into bootstrapping code

pgebheim committed:
Make mock mesh server send heartbeats

pgebheim committed:
Move browser mesh code into its own module

pgebheim committed:
Checkpoint: Rewire ZeroX setup

pgebheim committed:
Update single-thread-connector mocking for new function

pgebheim committed:
Fix ContractEvents.test.ts

ContractEvents now requires that the address for logs be registered or
we throw an error in order to help with bad deploys and ABI generation.
Updated the test to register its test address so that there is no error.

pgebheim committed:
Use property getter consistently

pgebheim committed:
Pass account through API

pgebheim committed:
Reformatting

pgebheim committed:
Only start ZeroX if config has the section present

pgebheim committed:
Use jest-expect interface for rejects

pgebheim committed:
Merge remote-tracking branch 'origin/master' into pgebheim/bootrapping+master

pgebheim committed:
Update flash foe new bootstrapping configs

pgebheim committed:
Fix issues preventing UI from loading

pgebheim committed:
Check

pgebheim committed:
Merge remote-tracking branch 'origin/master' into pgebheim/refactor-connectors-and-bootrapping

pgebheim committed:
Filter out cancel events with asset data that we don't want to see

pgebheim committed:
Merge remote-tracking branch 'origin/master' into pgebheim/refactor-connectors-and-bootrapping

robert-e-davidson3 commented on packages/augur-sdk/src/state/Server.ts:
```diff
@@ -1,13 +1,23 @@
+    ethereum: {
+      http: settings.ethNodeURLs[4]
+    },
+    syncing: {
```
The type marks this as optional. Can it be removed?

pgebheim replied:
`syncing` can be removed and it will use defaults.

Probably better to eschew defaults...

robert-e-davidson3 commented on packages/augur-sdk/src/state/db/ZeroXOrders.ts:
```diff
@@ -120,20 +120,24 @@ export class ZeroXOrders extends AbstractTable {
-    return this.augur.zeroX.subscribeToMeshEvents(this.handleMeshEvent.bind(this));
+  subscribeToOrderEvents() {
+    // This only works if `zeroX` has been set on the augur instance and
+    // it doesn't get over-written so... something.
```
Should `zeroX` be hidden behind getter-setter interfaces so overwrite can be handled?

pgebheim replied:
The getter/setter interface doesn't work in this direction.

the `ZeroX` instance is a server-side class, an it basically subscribes to the the mesh events of the things it owns. On the server-side this is the rpcClient. I need to add another layer to proxy events from the Client side to the Server side over the connector for the BrowserMesh.

Once that's the case the `mesh` instance on `ZeroX` should be a wrapper object which actually listens to global events an then re-emits them. Or that's my thinking at a least.

Currently, when just implemented with the SingleThreadConnector it's using a real BrowserMesh instance as the event source.

robert-e-davidson3 commented on packages/augur-sdk/src/state/test-connector.ts:
```diff
@@ -1,15 +1,11 @@
 // An example how to use Augur to retrieve data
```
Unrelated to the changes here: should this file even exist? Maybe it should be embedded in a markdown document? If it's just an example then we shouldn't be shipping it with our sdk code. Especially if importing it would start up a hardcoded server.

pgebheim replied:
Yeah -- currently this is useful because it forces us to keep it up to date (as it builds with tsc -b). Perhaps an examples directory?

robert-e-davidson3 replied:
Ah interesting. Yeah makes sense.

robert-e-davidson3 commented on packages/augur-test/src/libs/MockMeshServer.ts:
```diff
@@ -87,22 +88,41 @@ export class MockMeshServer {
+
+        if (params[0] === "heartbeat") {
+            subscription = heartbeatSub;
+            this._subInterval = setInterval(() => {
```
If we mess up and subscribe twice then this will create a second interval function that can't be cleared.

pgebheim replied:
Good call -- I can error if it's being called twice.

robert-e-davidson3 commented on packages/augur-test/src/tests/api/Gnosis.test.ts:
```diff
@@ -189,6 +189,7 @@ describe('Gnosis :: ', () => {
             status: GnosisSafeState.AVAILABLE,
           });
 
+          console.log(11);
```
```suggestion
```

robert-e-davidson3 commented on packages/augur-test/src/tests/api/ZeroX.test.ts:
```diff
@@ -53,6 +53,8 @@ describe('Augur API :: ZeroX :: ', () => {
       const johnConnector = new Connectors.DirectConnector();
       const johnGnosis = new MockGnosisRelayAPI();
       john = await ContractAPI.userWrapper(ACCOUNTS[0], provider, addresses, johnConnector, johnGnosis, meshClient, meshBrowser);
+      expect(john).not.toBeUndefined();
```
```suggestion
      expect(john).toBeDefined();
```

pgebheim committed:
Protect mock relayer from double subscribe issues

robert-e-davidson3 commented on packages/augur-test/src/tests/connector/ws-connector.test.ts:
```diff
@@ -38,8 +40,20 @@ jest.mock('websocket-as-promised', () => {
+    sdk: {
+      ws: 'ws://localhost:9001'
+    },
+    syncing: {
```
optional?

pgebheim committed:
Update packages/augur-test/src/tests/api/ZeroX.test.ts

Co-Authored-By: Robert E. Davidson III <45945043+robert-e-davidson3@users.noreply.github.com>

robert-e-davidson3 commented on packages/augur-test/src/tests/connector/ww-connector.test.ts:
```diff
@@ -34,9 +34,6 @@
 // jest.mock("@augurproject/sdk/build/state/index", () => {
 //   return {
 //     __esModule: true,
-//     buildAPI: () => {
-//       // return new API(john.augur, db);
-//     },
```
Delete the whole file maybe?

pgebheim replied:
oh heh the whole test is commented out....

I think we should fix it.

pgebheim committed:
Update packages/augur-test/src/tests/api/Gnosis.test.ts

Co-Authored-By: Robert E. Davidson III <45945043+robert-e-davidson3@users.noreply.github.com>

nuevoalex commented on packages/augur-sdk/src/api/ContractEvents.test.ts:
```diff
@@ -85,8 +85,48 @@ test('get event topics', async () => {
   const provider = makeProviderMock({ eventTopic });
 
-  const contractEvents = new ContractEvents(provider, '0x0', '0x0', '0x0', '0x0');
+  const contractEvents = new ContractEvents(provider, '0x0there', '0x0', '0x0', '0x0');
```
0x0there ?

pgebheim replied:
Not needed in this test, actually. The mocks use a totally fake address of `0xthere` --- See line 70 of the same file.

robert-e-davidson3 commented on packages/augur-ui/src/services/augursdk.ts:
```diff
@@ -1,115 +1,85 @@
+        });
       }
+      this.sdk = await createClient(config, connector, account, signer, provider, enableFlexSearch);
+      // This is messy -- that is neededing the connector to know about `sdk`
```
```suggestion
      // This is messy -- that is needing the connector to know about `sdk`
```

pgebheim committed:
Move the browser mesh wrappers into the UI

pgebheim committed:
Remove accessor specifiers from non-ctor function

pgebheim committed:
Add disconnect back

robert-e-davidson3 committed:
various fixes. mostly typing and lint. also got single-thread-connector to pass

robert-e-davidson3 committed:
fix setAddresses

pgebheim committed:
Fix

pgebheim committed:
Re-organize how configs happen for connector's client

pgebheim committed:
Pass in 0x endpoint and fix typo for STC

pgebheim committed:
Remove RPC port from bootstrap

pgebheim committed:
Merge branch 'master' into pgebheim/refactor-connectors-and-bootrapping

robert-e-davidson3 committed:
bind handleMergeEvent

robert-e-davidson3 committed:
Revert "bind handleMergeEvent"

This reverts commit 8a153db169c6e68b0bca0eae60e0f0d42d7e0824.

robert-e-davidson3 committed:
binding handleMergeEvent

pgebheim committed:
Remove spamming

joeykrug committed:
fixed a react error

joeykrug committed:
updated trade interval w/ new alex changes

pgebheim committed:
Only emit RPC events if no mesh, only emit mesh events if nonempty

pgebheim committed:
Merge branch 'pgebheim/refactor-connectors-and-bootrapping' of github.com:AugurProject/augur into pgebheim/refactor-connectors-and-bootrapping

pgebheim committed:
Fix

pgebheim committed:
MockMoshServer support for handling port conflicts
Only reconnect on initial error

robert-e-davidson3 committed:
mock db now has 0x enabled

pgebheim committed:
Disable mesh

pgebheim committed:
Make ZeroXOrders table if mesh OR rpc is enabled

joeykrug committed:
Update one-docker-to-rule-them-all.sh

joeykrug committed:
updated order exp stuff

joeykrug committed:
fixed order sorting in zerox and added looplimit and passing a list of attempted orders

pgebheim committed:
Switch to create-canned-markets during pop dockers build

pgebheim committed:
Update ZeroX Gnosis Tests

Gnosis Relayer is *required* in our code paths, but its turned off by
setting the useSafe property of the ContractDependenciesGnosis object
owned by the API to false.

pgebheim committed:
Tell http server to close to stop tests hangs

joeykrug committed:
fixes 5326

joeykrug committed:
Merge branch 'pgebheim/refactor-connectors-and-bootrapping' of https://github.com/AugurProject/augur into pgebheim/refactor-connectors-and-bootrapping

pgebheim committed:
Merge branch 'pgebheim/refactor-connectors-and-bootrapping' of github.com:AugurProject/augur into pgebheim/refactor-connectors-and-bootrapping
